File: Leetcode-python/+Problems/+Interview/twoSum.py
'''
Two sum
'''

def twoSum(arr, sum):
    map = {}

    for i in range(0, len(arr)):
        x = arr[i]
        goal = sum - x

        if goal in map:
            return True
        else:
            map[x] = i

if __name__ == "__main__":
    arr = [5,7,1,2,8,4,3]
    sum = 12    

    value = twoSum(arr, sum)
    print(value)

File: coding-interview-problems/Patterns/2-pointers/twoSum.go
package main

import "fmt"

func twoSum(arr []int, target int) []int {
	st := make(map[int]int)
	for i := range arr {
		comp := target - arr[i]
		if idx, ok := st[comp]; ok {
			return []int{idx, i}
		}
		st[arr[i]] = i
	}
	return nil
}

func main() {
	input := []int{2, 7, 11, 15}
	target := 9
	result := twoSum(input, target)
	fmt.Println(result)
}
